src/services/EdFi.DataManagementService.Api/Core/Validation/DocumentValidator.cs

```diff
@@ -14,32 +16,79 @@ public interface IDocumentValidator
+    public IEnumerable<string> Validate(PipelineContext context, ValidatorContext validatorContext)
     {
-        var formatValidationResult = documentBody.ValidateJsonFormat();
+        var formatValidationResult = context.FrontendRequest.Body.ValidateJsonFormat();
 
         if (formatValidationResult != null && formatValidationResult.Any())
         {
             return formatValidationResult;
         }
 
-        EvaluationOptions? validatorEvaluationOptions =
+        EvaluationOptions validatorEvaluationOptions =
             new() { OutputFormat = OutputFormat.List, RequireFormatValidation = true };
 
         var resourceSchemaValidator = schemaValidator.GetSchema(validatorContext);
-        var results = resourceSchemaValidator.Evaluate(documentBody, validatorEvaluationOptions);
+        var results = resourceSchemaValidator.Evaluate(
+            context.FrontendRequest.Body,
+            validatorEvaluationOptions
+        );
+
+        var pruneResult = PruneOverpostedData(context.FrontendRequest.Body, results);
+
+        if (pruneResult is PruneResult.Pruned pruned)
+        {
+            // Used pruned body for the remainder of pipeline
+            context.FrontendRequest = context.FrontendRequest with
+            {
+                Body = pruned.prunedDocumentBody
+            };
+
+            results = resourceSchemaValidator.Evaluate(
+                context.FrontendRequest.Body,
+                validatorEvaluationOptions
+            );
+        }
 
         return PruneValidationErrors(results);
 
-        List<string>? PruneValidationErrors(EvaluationResults results)
+        PruneResult PruneOverpostedData(JsonNode? documentBody, EvaluationResults evaluationResults)
+        {
+            if (documentBody == null)
+                return new PruneResult.NotPruned();
+
+            var additionalProperties = evaluationResults
+                .Details.Where(r =>
+                    r.EvaluationPath.Segments.Any() && r.EvaluationPath.Segments[^1] == "additionalProperties"
+                )
+                .ToList();
+
+            if (additionalProperties.Count == 0)
+                return new PruneResult.NotPruned();
+
+            foreach (var additionalProperty in additionalProperties)
+            {
+                JsonObject jsonObject = documentBody.AsObject();
+                var prunedJsonObject = jsonObject.RemoveProperty(
+                    additionalProperty.InstanceLocation.Segments
+                );
+                var prunedDocumentBody = JsonNode.Parse(prunedJsonObject.ToJsonString());
+                Trace.Assert(prunedDocumentBody != null, "Unexpected null after parsing pruned object");
+                documentBody = prunedDocumentBody;
+            }
+
+            return new PruneResult.Pruned(documentBody);
+        }
+
+        List<string> PruneValidationErrors(EvaluationResults results)
```

Comment:
I'd rename this. The `PruneOverpostedData` function is pruning overposted data, but `PruneValidationErrors` isn't pruning validation errors. I'd suggest `ValidationErrorsFrom` because line 61 will then read `return ValidationErrorsFrom(results);`